SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp

```diff
@@ -2121,22 +2203,46 @@ FRPCErrorInfo USpatialReceiver::ApplyRPCInternal(UObject* TargetObject, UFunctio
 		}
 
 		// Get the RPC target Actor.
+		const ERPCType RPCType = PendingRPCParams.Type;
+		const Worker_EntityId EntityId = PendingRPCParams.ObjectRef.Entity;
 		AActor* Actor = TargetObject->IsA<AActor>() ? Cast<AActor>(TargetObject) : TargetObject->GetTypedOuter<AActor>();
-		ERPCType RPCType = PendingRPCParams.Type;
+		bool bServerRPCType = (RPCType == ERPCType::ServerReliable || RPCType == ERPCType::ServerUnreliable);
 
-		if (Actor->Role == ROLE_SimulatedProxy && (RPCType == ERPCType::ServerReliable || RPCType == ERPCType::ServerUnreliable))
+		if (Actor->Role == ROLE_SimulatedProxy && bServerRPCType)
 		{
 			ErrorInfo.ErrorCode = ERPCResult::NoAuthority;
 			ErrorInfo.QueueProcessResult = ERPCQueueProcessResult::DropEntireQueue;
+
+			if (EventTracer->IsEnabled())
+			{
+				Worker_ComponentId ComponentId = RPCRingBufferUtils::GetRingBufferComponentId(RPCType);
+				RPCService->SpanIdCache.DropSpanIds(EntityComponentId(EntityId, ComponentId));
```

Comment:
Shouldn't this same span ID be associated with all rpcs processed that tick?
I'm not sure there is ever a reason to drop things before the end of the tick.

Reply:
We are not getting rid of the SpanId from the store as such, rather the link between the update to a component field and the SpanId. We have two SpanId caches now. One that takes worker ops and one that sits on the RPCService. The worker ops SpanId cache is cleared every tick. The RPC cache is only cleared when a component field change has been resolved. As RPCs can be queued, we might not resolve an RPC for many ticks. When we have processed or dropped and RPC we can remove the link between a component field change to a SpanId.